pyiron_base/database/filetable.py

```diff
@@ -56,21 +56,22 @@ class FileTableSingleton(ABCMeta):
     _instances = {}
 
     def __call__(cls, index_from):
-        _path = os.path.abspath(os.path.expanduser(index_from))
-        if _path not in cls._instances:
-            common_path = _get_most_common_path(
-                path=_path, reference_paths=cls._instances.keys()
+        path = os.path.abspath(os.path.expanduser(index_from))
+        if path not in cls._instances:
+            cls._instances[path] = super(FileTableSingleton, cls).__call__(
```

Comment:
@pmrv that is because it is confusing 😆 

Indeed, the `cls` variable means different things in the metaclass `__call__` signature and in the metaclass class method signatures. In `FileTableSingleton.__call__`, `cls` is the derived child class using that metaclass (in our case, `FileTable`); if you stick a class method on the metaclass, then `cls` is just the metaclass. This is why the `super().__call__(cls)` call didn't work inside the metaclass class method -- the super it was expecting  the derived class but got the metaclass.

Example:
```python
from abc import ABCMeta

class MyMeta(ABCMeta):
    def __call__(cls):
        print(f"In metaclass __call__, cls = {cls}({type(cls)})")
        cls.meta_class_method()
        cls.meta_method()
        return super().__call__()
    
    @classmethod
    def meta_class_method(meta_cls):
        print(f"In metaclass class method, meta_cls = {meta_cls}({type(meta_cls)})")
        
    def meta_method(cls):
        print(f"In metaclass method, cls = {cls}({type(cls)})")
        
        
class MyClass(metaclass=MyMeta):
    def __init__(self):
        print(f"In class __init__, self = {self}({type(self)})")
        
MyClass() 
>>>In metaclass __call__, cls = <class '__main__.MyClass'>(<class '__main__.MyMeta'>)
>>>In metaclass class method, meta_cls = <class '__main__.MyMeta'>(<class 'type'>)
>>>In metaclass method, cls = <class '__main__.MyClass'>(<class '__main__.MyMeta'>)
>>>In class __init__, self = <__main__.MyClass object at 0x14edec410>(<class '__main__.MyClass'>)
```

IMO, `cls` is the right name for the first argument, but it is the _derived_ `cls` not the metaclass `cls`! If we ever needed an actual class method on a metaclass, I'd probably name this argument `meta_cls` (I couldn't quickly find on google if there's already some other standard for this).

Thus, the current implementation with the method on the metaclass using `cls` is consistent and, IMO, good. I believe the old implementation would have worked if it had been a regular method instead of class method, but I actually like this refactoring more anyhow.

I'll clean up the super calls as a plain `super()` should work.